src/enclave/Enclave/ExpressionEvaluation.h

```diff
@@ -961,6 +961,40 @@ class FlatbuffersExpressionEvaluator {
         result_is_null);
     }
 
+    case tuix::ExprUnion_Upper:
+    {
+      auto n = static_cast<const tuix::Upper *>(expr->expr());
+      const tuix::Field *str = flatbuffers::GetTemporaryPointer(builder, eval_helper(row, n->child()));
+
+      if (str->value_type() != tuix::FieldUnion_StringField) {
+        throw std::runtime_error(
+          std::string("tuix::Substring requires str String, not ")
+          + std::string("str ")
+          + std::string(tuix::EnumNameFieldUnion(str->value_type())));
+      }
+
+      // Obtain the input as a string
+      bool result_is_null = str->is_null();
```

Comment:
If the input field is NULL, then no further processing needs to be done and we can just return a NULL value.

Reply:
Added null parsing. Doesn't seem to be working with null test...

Reply:
Null test working. Null is null.asInstanceOf[String]